UI/MainScreen.cpp

```diff
@@ -722,48 +722,52 @@ void MainScreen::CreateViews() {
+		ScrollView *scrollHomebrew = new ScrollView(ORIENT_VERTICAL, new LinearLayoutParams(FILL_PARENT, WRAP_CONTENT));
 
-	Choice *hbStore = tabHomebrew->HomebrewStoreButton();
-	if (hbStore) {
-		hbStore->OnClick.Handle(this, &MainScreen::OnHomebrewStore);
-	}
+		GameBrowser *tabAllGames = new GameBrowser(g_Config.currentDirectory, true, &g_Config.bGridView2,
+			mm->T("How to get games"), "http://www.ppsspp.org/getgames.html", 0,
+			new LinearLayoutParams(FILL_PARENT, FILL_PARENT));
+		GameBrowser *tabHomebrew = new GameBrowser(GetSysDirectory(DIRECTORY_GAME), false, &g_Config.bGridView3,
+			mm->T("How to get homebrew & demos", "How to get homebrew && demos"), "http://www.ppsspp.org/gethomebrew.html",
+			FLAG_HOMEBREWSTOREBUTTON,
+			new LinearLayoutParams(FILL_PARENT, FILL_PARENT));
 
-	scrollAllGames->Add(tabAllGames);
-	scrollHomebrew->Add(tabHomebrew);
+		Choice *hbStore = tabHomebrew->HomebrewStoreButton();
+		if (hbStore) {
+			hbStore->OnClick.Handle(this, &MainScreen::OnHomebrewStore);
+		}
 
-	leftColumn->AddTab(mm->T("Games"), scrollAllGames);
-	leftColumn->AddTab(mm->T("Homebrew & Demos"), scrollHomebrew);
+		scrollAllGames->Add(tabAllGames);
+		scrollHomebrew->Add(tabHomebrew);
 
-	tabAllGames->OnChoice.Handle(this, &MainScreen::OnGameSelectedInstant);
-	tabHomebrew->OnChoice.Handle(this, &MainScreen::OnGameSelectedInstant);
+		leftColumn->AddTab(mm->T("Games"), scrollAllGames);
+		leftColumn->AddTab(mm->T("Homebrew & Demos"), scrollHomebrew);
 
-	tabAllGames->OnHoldChoice.Handle(this, &MainScreen::OnGameSelected);
-	tabHomebrew->OnHoldChoice.Handle(this, &MainScreen::OnGameSelected);
+		tabAllGames->OnChoice.Handle(this, &MainScreen::OnGameSelectedInstant);
+		tabHomebrew->OnChoice.Handle(this, &MainScreen::OnGameSelectedInstant);
 
-	tabAllGames->OnHighlight.Handle(this, &MainScreen::OnGameHighlight);
-	tabHomebrew->OnHighlight.Handle(this, &MainScreen::OnGameHighlight);
+		tabAllGames->OnHoldChoice.Handle(this, &MainScreen::OnGameSelected);
+		tabHomebrew->OnHoldChoice.Handle(this, &MainScreen::OnGameSelected);
 
-	if (g_Config.recentIsos.size() > 0) {
-		leftColumn->SetCurrentTab(0);
-	} else if (g_Config.iMaxRecent > 0) {
-		leftColumn->SetCurrentTab(1);
-	}
+		tabAllGames->OnHighlight.Handle(this, &MainScreen::OnGameHighlight);
+		tabHomebrew->OnHighlight.Handle(this, &MainScreen::OnGameHighlight);
 
-	if (backFromStore_ || showHomebrewTab) {
-		leftColumn->SetCurrentTab(2);
-		backFromStore_ = false;
-		showHomebrewTab = false;
+		if (g_Config.recentIsos.size() > 0) {
+			leftColumn->SetCurrentTab(0);
+		} else if (g_Config.iMaxRecent > 0) {
+			leftColumn->SetCurrentTab(1);
+		}
+
+		if (backFromStore_ || showHomebrewTab) {
+			leftColumn->SetCurrentTab(2);
+			backFromStore_ = false;
+			showHomebrewTab = false;
+		}
+	} else {
+		leftColumn->Add(new Button(mm->T("Give PPSSPP permission to access storage")))->OnClick.Handle(this, &MainScreen::OnAllowStorage);
```

Comment:
What do you think of:

```c++
		LinearLayout *leftMessage = new LinearLayout(ORIENT_VERTICAL, new LinearLayoutParams(Margins(10)));
		leftMessage->Add(new TextView(mm->T("PPSSPP needs write access to save and run games")));
		leftColumn->Add(leftMessage);
		leftColumn->Add(new Button(mm->T("Grant PPSSPP storage permission"), new LinearLayoutParams(Margins(40, 0))));
```

I was hoping that would center the button but it's not exactly...

-[Unknown]